ykrt/src/mt.rs

```diff
@@ -321,13 +321,18 @@ impl MTThread {
                     } else {
                         let new_state = PHASE_COUNTING | ((count + 1) << PHASE_NUM_BITS);
                         let old_lp = lp;
-                        lp = loc.state.compare_and_swap(lp, new_state, Ordering::Release);
+                        lp = loc.state.compare_and_swap(lp, new_state, Ordering::Relaxed);
                         if lp == old_lp {
                             return None;
                         }
-                    }
```

Comment:
I wasn't expecting this hunk. Only changes to the memory orderings. Is it related?

Reply:
Possibly not -- but given that `PHASE_COUNTING` doesn't need to sync on anything other than the `Location`, `Release` was misleading.

Reply:
OK